mount/mount_linux.go

```diff
@@ -410,14 +383,7 @@ func (mounter *SafeFormatAndMount) formatAndMountSensitive(source string, target
 
 		if !readOnly {
 			// Run check tools on the disk to fix repairable issues, only do this for formatted volumes requested as rw.
-			var err error
-			switch existingFormat {
-			case "xfs":
-				err = mounter.checkAndRepairXfsFilesystem(source)
-			default:
-				err = mounter.checkAndRepairFilesystem(source)
-			}
-
+			err := mounter.checkAndRepairFilesystem(source)
```

Comment:
The code that is part of `checkAndRepairFilesystem(...)` used to be called before `// Check if the disk is already formatted`. As part of https://github.com/kubernetes/utils/pull/126 it was moved down here. Should we revert this change as well or is it necessary for some reason?

Reply:
We could revert that change too but I figured it is harmless and helps to keep diff of this PR small. We will have to fix all the breaking tests because order in which `fsck` gets called will change. 

Also it is perfectly fine to *not* run `fsck` on a unformatted disk.

Reply:
Ack. Sounds good to me. @msau42 @xing-yang @jsafrane any objections?

Reply:
Sounds good to me.

Reply:
lgtm. The bug report had the error past the fs format check, so it's not the fs format check that's failing.

Reply:
> The bug report had the error past the fs format check, so it's not the fs format check that's failing.

@gnufied can you confirm?

Reply:
which bug report? https://github.com/kubernetes/utils/issues/141 or https://github.com/kubernetes/utils/issues/125 ? I assume #141 - yes the problem isn't format checking but problem was that repairing the disk actually prevented volume from being mounted and recovery (although in some cases XFS volume can't recover itself and requires `xfs_repair`).

So, the only valid reason I did not do a full revert is because I liked not having to do a `fsck` if disk is unformatted.

Reply:
Ack. SGTM